packages/server/app/Models/User.ts

```diff
@@ -47,6 +49,78 @@ export default class User extends BaseModel {
     public mpeRooms: ManyToMany<typeof MpeRoom>;
     ///
 
+    @belongsTo(() => SettingVisibility, {
+        localKey: 'uuid',
+        foreignKey: 'playlistsVisibilitySettingUuid',
+    })
+    public playlistsVisibilitySetting: BelongsTo<typeof SettingVisibility>;
+
+    @column({ columnName: 'playlists_visibility_setting_uuid' })
+    public playlistsVisibilitySettingUuid: string;
+
+    @beforeCreate()
+    public static async assignPlaylistsVisibilitySetting(
+        user: User,
+    ): Promise<void> {
+        if (user.playlistsVisibilitySettingUuid === undefined) {
+            const publicVisibilitySetting =
+                await SettingVisibility.findByOrFail(
+                    'name',
+                    UserSettingVisibility.enum.PUBLIC,
+                );
+
+            user.playlistsVisibilitySettingUuid = publicVisibilitySetting.uuid;
+        }
+    }
+
+    @belongsTo(() => SettingVisibility, {
```

Comment:
BelongsTo is quite disturbing here but you're right the foreignKey being in this model no other way to deal with it haha